docs/deployment/k8s-installation.md

```diff
@@ -0,0 +1,280 @@
+# Installation to non-local Kubernetes
+
+This guide will walk you through the steps to install Racetrack on a non-local Kubernetes cluster,
+such as AKS, GKE, EKS or a self-hosted Kubernetes.
+
+## Prerequisites
+
+1. Install [kubectl](https://kubernetes.io/docs/tasks/tools/) (version 1.24.3 or higher)
+
+## Create a Kubernetes cluster
+
+The first step is to create a Kubernetes cluster.
+Let's assume you have already created an
+[AKS cluster on Azure](https://learn.microsoft.com/en-us/azure/aks/learn/quick-kubernetes-deploy-cli#create-aks-cluster),
+and you have access to it using the `kubectl` tool.
+Verify the connection to your cluster using the `kubectl get nodes` command.
+
+Next, set this cluster as the default one:
+```sh
+kubectl config get-contexts
+kubectl config use-context aks-racetrack # k8s context is `aks-racetrack` in this tutorial
+kubectl config set-context --current --namespace=racetrack
+```
+
+## Prepare Docker Registry
+Racetrack needs a Docker registry to store the images of the jobs.
+We need to instruct Kubernetes to pull images from there.
+
+Let's assume we have a Docker registry at `ghcr.io/theracetrack/racetrack/` with
```

Comment:
I'm afraid Racetrack still needs a Docker registry. While we could use the `ghcr.io` and we could push the **base** images beforehand so that Racetrack sees they're already there and that's okay. However, in the next step Racetrack builds the **job** image which is always unique and has a unique tag composed of a current timestamp. Thus, it always needs the registry to deploy jobs. Right now, it's not supported in Racetrack as it always pushes at least one image.

Plus, it might be surprising for the users that they can only deploy samples in this scenario. Out of curiosity, they would try to deploy something else, and quickly they discover that such Racetrack is useless.

Reply:
That's fair.